### DISCUSSION TITLE
Rationale for InverseVolumeRamp

### DISCUSSION THREAD

**Discussion by cslominski** [glotzerlab/hoomd-blue#1814] [General]
Hi,

When compressing a monodisperse suspension using InverseVolumeRamp, the volume fraction \phi changes like d \phi / d t = constant. In the common Lubachevsky-Stillinger (LS) algorithm which instead grows the particle diameters linearly with time, d \phi / d t ~ t^2. To me, the InverseVolumeRamp algorithm makes more sense from a thermodynamic perspective; however, I was not able to find computational studies of the effects of its compression rate on the suspension structure. There are many such studies available for the LS algorithm.

I have two questions:
1) What was your rationale for the implementation of InverseVolumeRamp?
2) Could you please point me to references using either the InverseVolumeRamp algorithm or others that are equivalent?

Thanks.

**Comment by tcmoore3** [comment]
The `InverseVolumeRamp` on its own is no algorithm; it is just a trivial function that takes a time and yields boxes where 1/(volume of box) varies linearly in time. The motivation for implementing it was we wanted to slowly compress hard particle systems. Many publications on hard particle systems use a _quick compression_ algorithm to quickly get the system to a target density, but I'm not aware of any references to a slow compression of a hard particle system.

**Reply by joaander** [reply]
To add to this: You can implement any functional form of a volume ramp with a custom variant. If you think there is a particular volume ramp that is of interest to the HOOMD-blue user community, feel free to write a C++ implementation and contribute by pull request.

**Reply by cslominski** [reply]
OK thanks guys. If I ever add a custom ramp I'll submit a pull request for it. If you or others come across studies compressing with InverseVolumeRamp or equivalent, please post them here. If I publish with it, I will try to remember to post it here, although I won't be looking at jamming etc.
